zerver/tests/test_auth_backends.py

```diff
@@ -1026,6 +1029,10 @@ def test_social_auth_registration_existing_account(self) -> None:
     def stage_two_of_registration(self, result: HttpResponse, realm: Realm, subdomain: str,
                                   email: str, name: str, expected_final_name: str,
                                   skip_registration_form: bool,
+                                  expected_social_avatar_content_path: str='',
+                                  user_avatar_url: Optional[str]='',
+                                  use_social_avatar: Optional[bool]=False,
+                                  avatar_upload_error: Optional[bool]=False,
```

Comment:
This function is ugly and with lots of arguments already, I think adding `avatar_upload_error` isn't worth making it even worse :( I wonder if it wouldn't be better to just mock `upload_avatar_image_from_url` to raise the desired exception in the test in which you want to test the error, and avoid adding this extra stuff to `stage_two_of_registration` :thinking:

Reply:
Oh and these type annotations aren't correct:
```
                                  user_avatar_url: Optional[str]='',
                                  use_social_avatar: Optional[bool]=False,
                                  avatar_upload_error: Optional[bool]=False,
```

none of these are actually `Optional`